cypress/e2e/cloud/pinned.test.ts

```diff
@@ -0,0 +1,330 @@
+      'contain.text',
+      'Pin a task, dashboard, or notebook here'
+    )
+  })
+
+  describe('Pin dashboard tests', () => {
+    beforeEach(() => {
+      cy.createDashboard(orgID).then(() => {
+        cy.setFeatureFlags({
+          pinnedItems: true,
+          docSearchWidget: true,
+        }).then(() => {
+          cy.fixture('routes').then(({orgs}) => {
+            cy.visit(`${orgs}/${orgID}/dashboards-list`)
+            cy.getByTestID('tree-nav')
+          })
+        })
+      })
+    })
+    it('pins a dashboard to the homepage for easy access as a pinned item', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-pin-menu').click()
+          cy.getByTestID('context-pin-dashboard').click()
+        })
+      cy.visit('/')
+      cy.getByTestID('pinneditems--container').within(() => {
+        cy.getByTestID('pinneditems--card').should('exist')
+      })
+    })
+
+    it('reflects an edit to the dashboard name on the dashboard card', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-pin-menu').click()
+          cy.getByTestID('context-pin-dashboard').click()
+        })
+      cy.getByTestID('dashboard-card').within(() => {
+        cy.getByTestID('dashboard-card--name')
+          .first()
+          .trigger('mouseover')
+
+        cy.getByTestID('dashboard-card--name-button')
+          .first()
+          .click()
+
+        cy.get('.cf-input-field')
+          .type('Bucks In Six')
+          .type('{enter}')
+      })
+
+      cy.visit('/')
+      cy.getByTestID('pinneditems--container').within(() => {
+        cy.getByTestID('pinneditems--link').should(
+          'contain.text',
+          'Bucks In Six'
+        )
+      })
+    })
+
+    it('unpins a card which removes it from the pinned list', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-pin-menu').click()
+          cy.getByTestID('context-pin-dashboard').click()
+        })
+      cy.visit('/')
+      cy.getByTestID('pinneditems--card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('pinneditems-delete--menu').click()
+          cy.getByTestID('pinneditems-delete--confirm').click()
+        })
+      cy.getByTestID('pinneditems--emptystate').should(
+        'contain.text',
+        'Pin a task, dashboard, or notebook here'
+      )
+    })
+
+    it('unpins when the underlying resource is removed', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-delete-menu').click()
+          cy.getByTestID('context-delete-dashboard').click()
+        })
+
+      cy.visit('/')
```

Comment:
I would add a `cy.getByTestID('nav-tree')` here after the `cy.visit('/')` so the test waits for the page to load before continuing. Should reduce flakes.